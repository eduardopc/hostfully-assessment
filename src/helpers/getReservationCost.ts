import { differenceInDays } from "date-fns";
import { GENERAL } from "languages";

import { PlaceSelectOption } from "types";

type GetReservationCostProps = {
  dates?: string[];
  pricePerDay: PlaceSelectOption["pricePerDay"];
};

export const getReservationCost = ({
  dates,
  pricePerDay,
}: GetReservationCostProps): string => {
  if (!dates || !pricePerDay) return GENERAL.remake_reservation;

  const days = differenceInDays(dates[1], dates[0]);

  return `$${days * pricePerDay}`;
};
