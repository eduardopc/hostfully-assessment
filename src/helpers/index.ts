export { isBookingDateRangeAvailable } from "./isBookingDateRangeAvailable";
export { showToastMessage } from "./toast";
