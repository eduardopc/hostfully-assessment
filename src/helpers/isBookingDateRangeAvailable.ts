import { isWithinInterval } from "date-fns";

import { BookingDataProps } from "contexts/types";

export const isBookingDateRangeAvailable = (
  from: Date,
  to: Date,
  bookings: BookingDataProps[]
): boolean => {
  return !bookings.some(
    (booking) =>
      isWithinInterval(from, {
        start: booking.date[0].from,
        end: booking.date[0].to,
      }) ||
      isWithinInterval(to, {
        start: booking.date[0].from,
        end: booking.date[0].to,
      })
  );
};
