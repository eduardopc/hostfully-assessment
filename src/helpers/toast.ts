import { toast, TypeOptions } from "react-toastify";

type ShowToastMessage = {
  message: string;
  type: TypeOptions;
  onOpen?: () => void;
};

export const showToastMessage = ({
  message,
  type,
  onOpen,
}: ShowToastMessage) => {
  toast[type](message, {
    position: "top-right",
    autoClose: 5000,
    hideProgressBar: true,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: false,
    theme: "colored",
    onOpen: () => {
      onOpen?.();
    },
  });
};
