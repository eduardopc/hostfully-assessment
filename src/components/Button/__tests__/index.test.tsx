import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";
import { AddShoppingCart } from "@styled-icons/material-outlined/AddShoppingCart";

import { Button } from "..";

describe("<Button />", () => {
  it("should render the medium size by default ", () => {
    const { container } = renderWithTheme(<Button>Hostfully</Button>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render an icon version", () => {
    renderWithTheme(
      <Button icon={<AddShoppingCart data-testid="icon" />}>Hostfully</Button>
    );

    expect(screen.getByText(/Hostfully/i)).toBeDefined();
    expect(screen.getByTestId("icon")).toBeDefined();
  });

  it("should render Button as a link", () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Hostfully
      </Button>
    );

    expect(screen.getByRole("link", { name: /Hostfully/i })).toHaveProperty(
      "href",
      "http://localhost:3000/link"
    );
  });
});
