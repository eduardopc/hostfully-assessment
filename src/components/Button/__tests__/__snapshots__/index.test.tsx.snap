// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render the medium size by default  1`] = `
<button
  class="sc-bgqQPU ibbghM"
>
  <span>
    Hostfully
  </span>
</button>
`;
